### PR TITLE
ns2 parser

### PR DESCRIPTION
**Pull request by grimmfl**

### CODE REVIEW

File: pons/routing/router.py [hunk 1]
```diff
@@ -98,10 +103,28 @@ def on_scan_received(self, msg: pons.Message, remote_id: int):
...
+            if msg.dst == self.my_id:
+                # print("msg arrived", self.my_id)
+                self.netsim.routing_stats['delivered'] += 1
+                self.netsim.routing_stats['hops'] += msg.hops
+                self.netsim.routing_stats['latency'] += self.env.now - msg.created
+                self.netsim.event_manager.on_message_delivered(self.my_id, remote_id, msg)
+            else:
+                # print("msg not arrived yet", self.my_id)
+                self.netsim.event_manager.on_message_received(self.my_id, remote_id, msg)
+                self.forward(msg)
```
Comment (gh0st42) [inline pons/routing/router.py]: the standard router should not have any biz logic, so forwarding on its own is a no go.
the question is how can this be refactored to automatically send out the events to the event manager but at the same time keep the logic in the actual routing impls.

File: pons/routing/router.py [hunk 2]
```diff
@@ -98,10 +103,28 @@ def on_scan_received(self, msg: pons.Message, remote_id: int):
...
     def on_peer_discovered(self, peer_id):
-        self.log("peer discovered: %d" % peer_id)
+        #self.log("peer discovered: %d" % peer_id)
+        self.netsim.event_manager.on_peer_discovery(self.my_id, peer_id)
 
     def on_msg_received(self, msg: pons.Message, remote_id: int):
-        self.log("msg received: %s from %d" % (msg, remote_id))
+        self.netsim.routing_stats['relayed'] += 1
+        if not self.is_msg_known(msg):
+            self.remember(remote_id, msg)
```
Comment (gh0st42) [inline pons/routing/router.py]: this is convinient for most routing algorithms but quite heavy for an empty router without logic.

File: pons/routing/router.py [hunk 3]
```diff
@@ -98,10 +103,28 @@ def on_scan_received(self, msg: pons.Message, remote_id: int):
...
+            msg.hops += 1
+            self.store_add(msg)
```
Comment (gh0st42) [inline pons/routing/router.py]: what if i want to model a router without storage, only directly forwarding and then forgetting the bundle?
here, i always have a persistent store. 
yes, i could ignore it but then the question is does it have performance impact for evaluations?

File: pons/routing/router.py (hunk 1 above)
Comment (gh0st42) [reply pons/routing/router.py]: but then again router impl get so much shorter, not sure what the most elegant solution is.

**Review by gh0st42 — CHANGES_REQUESTED** [review]
thanks for your PR.
it would be nice to have working examples and example ns2 movement traces to load in here. or a unittest with a mapping between the one and ns2 movement or visually as plots to see that the patterns are roughly the same

File: tests/__init__.py
Comment (gh0st42) [inline tests/__init__.py:1]: maybe a README.md in the tests folder describing what can be found here would be helpful.
same with the mobility subfolder, why the 3 movement traces? what edge cases do they cover?

File: tests/mobility/ns2_tests.py
Comment (gh0st42) [inline tests/mobility/ns2_tests.py:1]: some documentation for endusers how to run the tests or a test.sh script in the root folder might be nice

**Review by gh0st42 — CHANGES_REQUESTED** [review]

**Review by gh0st42 — COMMENTED** [review]
looks good
